src/Methods/PaymentOrderValidator.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace Payone\Methods;
+
+use Payone\Adapter\Logger;
+use Payone\Helpers\AddressHelper;
+use Payone\Services\SettingsService;
+use Plenty\Modules\Authorization\Services\AuthHelper;
+use Plenty\Modules\Order\Contracts\OrderRepositoryContract;
+use Plenty\Modules\Order\Models\Order;
+
+class PaymentOrderValidator
+{
+    /**
+     * @var AddressHelper
+     */
+    private $addressHelper;
+
+    /**
+     * @var Logger
+     */
+    private $logger;
+
+    /**
+     * PaymentValidator constructor.
+     *
+     * @param AddressHelper $addressHelper
+     * @param Logger $logger
+     */
+    public function __construct(AddressHelper $addressHelper, Logger $logger)
+    {
+        $this->addressHelper = $addressHelper;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param PaymentAbstract $payment
+     *
+     * @return bool
+     */
+    public function validate(PaymentAbstract $payment, SettingsService $settingsService, int $orderId)
+    {
+        /** @var OrderRepositoryContract $orderRepositoryContract */
+        $orderRepositoryContract = pluginApp(OrderRepositoryContract::class);
+        /** @var AuthHelper $authHelper */
+        $authHelper = pluginApp(AuthHelper::class);
+        /** @var Order $order */
+        $order = $authHelper->processUnguarded(
+            function () use ($orderRepositoryContract, $orderId) {
+                return $orderRepositoryContract->findOrderById($orderId);
```

Comment:
As this method is deprecated since 04.2021, use the new method:
```suggestion
                return $orderRepositoryContract->findById($orderId, ['amounts', 'addresses']);
```